dev/how-to/maintain/identify-slow-queries.md

```diff
@@ -1,256 +1,142 @@
-+-------------+------------------------------------------------------------------+----------------+
-| 0.676408014 | select t0.c0, t1.c1 from t_slim t0, t_wide t1 where t0.c0=t1.c1; | test@127.0.0.1 |
-+-------------+------------------------------------------------------------------+----------------+
-1 row in set
-Time: 0.014s
+SELECT query
+FROM information_schema.slow_query
+WHERE digest = ?;
 ```
 
-### 根据 SQL 指纹来查询类似 SQL 的慢查询
+是否有索引可用需要根据表结构和 SQL 查询具体分析，这里不再赘述
 
-如果查询了 TopN 的SQL 后，想查询相同 SQL 指纹的查询，可以用指纹作为过滤条件。
-
-```sql
-tidb > select query_time, query,digest from INFORMATION_SCHEMA.`SLOW_QUERY` where `Is_internal`=false order by `Query_time` desc limit 1;
-+-------------+-----------------------------+------------------------------------------------------------------+
-| query_time  | query                       | digest                                                           |
-+-------------+-----------------------------+------------------------------------------------------------------+
-| 0.302558006 | select * from t1 where a=1; | 4751cb6008fda383e22dacb601fde85425dc8f8cf669338d55d944bafb46a6fa |
-+-------------+-----------------------------+------------------------------------------------------------------+
-1 row in set
-Time: 0.007s
-tidb > select query, query_time from INFORMATION_SCHEMA.`SLOW_QUERY` where digest="4751cb6008fda383e22dacb601fde85425dc8f8cf669338d55d944bafb46a6fa";
-+-----------------------------+-------------+
-| query                       | query_time  |
-+-----------------------------+-------------+
-| select * from t1 where a=1; | 0.302558006 |
-| select * from t1 where a=2; | 0.401313532 |
-+-----------------------------+-------------+
-2 rows in set
-```
+### 优化器异常
 
-### 查询统计信息是 pseudo 的慢查询
+优化器异常通常是统计信息过时导致的。统计信息过时后，TiDB 优化器会使用一个基于经验假设的代价估算逻辑对表上的查询进行代价估算。可以通过如下查询发现那些统计信息过时导致的慢查询：
 
 ```sql
-tidb > select query, query_time, stats from INFORMATION_SCHEMA.`SLOW_QUERY` where is_internal=false and stats like('%pseudo%');
-+-----------------------------+-------------+---------------------------------+
-| query                       | query_time  | stats                           |
-+-----------------------------+-------------+---------------------------------+
-| select * from t1 where a=1; | 0.302558006 | t1:pseudo                       |
-| select * from t1 where a=2; | 0.401313532 | t1:pseudo                       |
-| select * from t1 where a>2; | 0.602011247 | t1:pseudo                       |
-| select * from t1 where a>3; | 0.50077719  | t1:pseudo                       |
-| select * from t1 join t2;   | 0.931260518 | t1:407872303825682445,t2:pseudo |
-+-----------------------------+-------------+---------------------------------+
+SELECT digest,
+       count(*) AS no_index_count
+FROM information_schema.slow_query
+WHERE stats like "%pseudo%"
+  AND is_internal = false
+GROUP BY digest
+ORDER BY no_index_count DESC
+LIMIT 10;
```

Comment:
How about those slow queries caused by stale stats which is not pseudo? should we mention the `show stats_meta` here?